kernel/src/thread/thread_table.rs

```diff
@@ -1,21 +1,30 @@
 // SPDX-License-Identifier: MPL-2.0
 
+use keyable_arc::KeyableArc;
+
 use super::{Thread, Tid};
-use crate::prelude::*;
+use crate::{prelude::*, process::posix_thread::PosixThreadExt};
+
+static POSIX_THREAD_TABLE: SpinLock<BTreeMap<Tid, Arc<Thread>>> = SpinLock::new(BTreeMap::new());
+static KERNEL_THREAD_TABLE: SpinLock<BTreeSet<KeyableArc<Thread>>> = SpinLock::new(BTreeSet::new());
```

Comment:
I suggest removing `KERNEL_THREAD_TABLE` for three reasons:

1. This table does not make sense. If a user already holds a `KeyableArc<Thread>`, which is same as `Arc<Thread>`, why does she want to query this table? A thread table is only useful when the key is TIDs so that a user can use TIDs to retrieve `Arc<Thread>`.
2. This table is not used by any one. The module does not provide a getter API.
3. This table introduces overheads when spawning and exiting threads, but we are now working hard to squeeze every bit of performance.

Reply:
The kernel thread table has been removed. 

Initially, `Task` only held a `Weak<Thread>`. For POSIX threads, the Process held an `Arc<Thread>`, but in the case of kernel threads, there was no owner for `Thread` without the kernel thread table.

To address this, I modified the logic so that `Task` now holds an `Arc<Thread>`, while `Thread` only holds a `Weak<Task>`. This ensures that if a task is running, the corresponding kernel thread will not be dropped. As a result, the `Process` will now hold an `Arc<Task>`. This adjustment also enhances the performance of the `current_thread` macro a bit.